src/client/lib/flux/flux.react.js

```diff
@@ -13,6 +14,11 @@ function now() {
 
 export default function flux(store) {
 
+  invariant(
+    typeof store === 'function',
+    `@flux(...): Store must be a pure function, ${typeof store} given`
+  );
+
```

Comment:
I prefer checking for `store` here since we can throw early with a nice error. Checking in constructor would've required a message like `new Flux(...): store must be...` which might be confusing for our users.